code/implementation/docker/machine/src/main/java/io/cattle/platform/docker/machine/api/machinhost/filter/MachineHostValidationFilter.java

```diff
@@ -0,0 +1,50 @@
+package io.cattle.platform.docker.machine.api.machinhost.filter;
+
+import static io.cattle.platform.docker.machine.constants.MachineHostConstants.*;
+import io.cattle.platform.core.model.PhysicalHost;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class MachineHostValidationFilter extends AbstractDefaultResourceManagerFilter {
+
+    private static final String DRIVER_CONFIG_EXACTLY_ONE_REQUIRED = "DriverConfigExactlyOneRequired";
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { MACHINE_KIND };
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { PhysicalHost.class };
```

Comment:
If you don't want to match the parent type "physicalHost" then don't return a class.  It's sufficient to return either something from getTypes or getTypeClasses, but both aren't required.